packages/gatsby/src/commands/develop.js

```diff
@@ -248,12 +249,25 @@ async function startServer(program) {
 module.exports = async (program: any) => {
   const detect = require(`detect-port`)
   const port =
-    typeof program.port === `string` ? parseInt(program.port, 10) : program.port
+    typeof program.port === `string`
+      ? parseInt(program.port, 10)
+      : program.port
 
   // Check if https is enabled, then create or get SSL cert.
   // Certs are named after `name` inside the project's package.json.
   if (program.https) {
     program.ssl = await getSslCert(program.sitePackageJson.name)
+  } else if (program[`cert-file`] && program[`key-file`]) {
```

Comment:
I am just thinking if this condition should go before `if (program.https)`.

Then if people would use `gatsby develop --https --cert-file=some_path --key-file=other_path` it would prioritze custom ssl "files" over generated one. Does this make sense?

Reply:
yeah it makes sense, my initial thought was that the precedence should go to https since that is the simpler and most likely to be used by most people. I can see arguments for both ways though

Reply:
yeah I can see both ways. I'll change it to whatever it needs to be to get it merged. Just let me know